roles/uperf/tasks/cleanup.yml

```diff
@@ -0,0 +1,55 @@
+---
+
+- block:
+  ### <POD> kind
+  - name: Get Server Pods
+    k8s_facts:
+      kind: Pod
+      api_version: v1
+      namespace: '{{ operator_namespace }}'
+      label_selectors:
+        - type = uperf-bench-server-{{ trunc_uuid }}
+    register: server_pods
+
+  - name: Pod names - to clean
+    set_fact:
+      clean_pods: |
+          [
+          {% for item in server_pods.resources %}
+            "{{ item['metadata']['name'] }}",
+          {% endfor %}
+          ]
+
+  - name: Cleanup run
```

Comment:
We must delete jobs rather than pods. if we don't do that jobs will be restarted by the kube job controller.
i.e
```
$ oc get job
oc get pod
NAME                                      COMPLETIONS   DURATION   AGE
uperf-client-10.131.0.74-102ba633         1/1           9m47s      5h8m
uperf-client-10.131.0.75-102ba633         1/1           9m48s      5h8m
uperf-client-10.131.0.76-102ba633         1/1           9m48s      5h8m
uperf-client-10.132.0.100-102ba633        1/1           9m47s      5h8m
uperf-client-10.132.0.101-102ba633        1/1           9m48s      5h8m
uperf-client-10.132.0.99-102ba633         1/1           9m48s      5h8m
uperf-client-10.133.0.100-102ba633        1/1           9m48s      5h8m
uperf-client-10.133.0.98-102ba633         1/1           9m48s      5h8m
uperf-client-10.133.0.99-102ba633         1/1           9m48s      5h8m
uperf-server-ip-10-0-149-84-0-102ba633    0/1           5h9m       5h9m
uperf-server-ip-10-0-149-84-1-102ba633    0/1           5h9m       5h9m
uperf-server-ip-10-0-149-84-2-102ba633    0/1           5h9m       5h9m
uperf-server-ip-10-0-187-173-0-102ba633   0/1           5h9m       5h9m
uperf-server-ip-10-0-187-173-1-102ba633   0/1           5h9m       5h9m
uperf-server-ip-10-0-187-173-2-102ba633   0/1           5h9m       5h9m
uperf-server-ip-10-0-220-135-0-102ba633   0/1           5h9m       5h9m
uperf-server-ip-10-0-220-135-1-102ba633   0/1           5h9m       5h9m
uperf-server-ip-10-0-220-135-2-102ba633   0/1           5h9m       5h9m
```

These jobs were created some hours ago  ^

The server pods were recreated (age 15m)
```
$ oc get pod
NAME                                            READY   STATUS      RESTARTS   AGE
benchmark-operator-58945b66b8-srzd6             3/3     Running     0          16m
uperf-client-10.131.0.74-102ba633-x69c8         0/1     Completed   0          5h8m
uperf-client-10.131.0.75-102ba633-5vx8m         0/1     Completed   0          5h8m
uperf-client-10.131.0.76-102ba633-q9lhh         0/1     Completed   0          5h8m
uperf-client-10.132.0.100-102ba633-qb29z        0/1     Completed   0          5h8m
uperf-client-10.132.0.101-102ba633-2tzd5        0/1     Completed   0          5h8m
uperf-client-10.132.0.99-102ba633-d2rqx         0/1     Completed   0          5h8m
uperf-client-10.133.0.100-102ba633-z4vft        0/1     Completed   0          5h8m
uperf-client-10.133.0.98-102ba633-p5rzc         0/1     Completed   0          5h8m
uperf-client-10.133.0.99-102ba633-9pvkp         0/1     Completed   0          5h8m
uperf-server-ip-10-0-149-84-0-102ba633-r2fbn    1/1     Running     0          15m
uperf-server-ip-10-0-149-84-1-102ba633-2kxgg    1/1     Running     0          15m
uperf-server-ip-10-0-149-84-2-102ba633-86t9h    1/1     Running     0          15m
uperf-server-ip-10-0-187-173-0-102ba633-5jmbn   1/1     Running     0          15m
uperf-server-ip-10-0-187-173-1-102ba633-gprm8   1/1     Running     0          15m
uperf-server-ip-10-0-187-173-2-102ba633-hg4nd   1/1     Running     0          15m
uperf-server-ip-10-0-220-135-0-102ba633-t5fwm   1/1     Running     0          15m
uperf-server-ip-10-0-220-135-1-102ba633-rj7qw   1/1     Running     0          15m
uperf-server-ip-10-0-220-135-2-102ba633-h4ggn   1/1     Running     0          15m
```

On the other hand, are we keeping the client pods?, is this expected?

Reply:
This clean up logic and behavior was of the master branch. I think it has progressed to this state unintentionally.  I am changing it so that the workload will cleanup the server pods (and server jobs), but leave the client pods around.  I have some personal tools that analyze the results for correctness. So leaving the client pods around has a purpose.